Seagate/cortx-utils
# CORTX-30213: Added Activity framework

# Problem Statement
- Add an infrastructure for maintaining information about various ongoing activities / tasks in the cluster and view them asynchronously.

# Design
-  For Bug describe the fix here. 
-  [For Feature, Post the link to the solution page on the confluence CORTX Foundation Library 
- https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/580550715/WIP+CORTX+Task+Framework](https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/580550715/WIP+CORTX+Task+Framework)

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing
- Are test cases updated along with code changes due to Enhancements/Bugs [Y/N]: Y
- Confirm that new test cases are added to regression and sanity plan files and relevant feature plan files [Y/N]:
- Confirm that Test Cases are added for new features added [Y/N]:  Y
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: Y
- Confirm Testing was performed with installed RPM/K8s deployment [Y/N]:  Y

# Review Checklist 
####  Before posting the PR please ensure:

# KVstore/Confstore feature (changes/fixes) checklist
#### Confirm changes are made for:

# Documentation
